pkg/network/ebpf/c/runtime/conntrack.c

```diff
@@ -0,0 +1,68 @@
+#include <linux/kconfig.h>
+#include <linux/version.h>
+#include <linux/types.h>
+
+#include "bpf_helpers.h"
+#include "bpf_endian.h"
+#include "conntrack.h"
+#include "conntrack-maps.h"
+#include "netns.h"
+#include "ip.h"
+
+#ifdef FEATURE_IPV6_ENABLED
+#include "ipv6.h"
+#endif
+
+#ifndef LINUX_VERSION_CODE
+# error "kernel version not included?"
+#endif
+
+SEC("kprobe/__nf_conntrack_hash_insert")
+int kprobe___nf_conntrack_hash_insert(struct pt_regs* ctx) {
+    struct nf_conn *ct = (struct nf_conn*)PT_REGS_PARM1(ctx);
+    u32 status = ct_status(ct);
+    if (status == 0) {
+        return 0;
+    }
+
+    struct nf_conntrack_tuple_hash tuplehash[IP_CT_DIR_MAX];
+    bpf_probe_read(&tuplehash, sizeof(tuplehash), &ct->tuplehash);
+
+    struct nf_conntrack_tuple orig = tuplehash[IP_CT_DIR_ORIGINAL].tuple;
+    struct nf_conntrack_tuple reply = tuplehash[IP_CT_DIR_REPLY].tuple;
+    if (!is_nat(&orig, &reply)) {
+        increment_telemetry_count(registers_dropped);
+        return 0;
+    }
+    u32 netns = get_netns(&ct->ct_net);
+    log_debug("kprobe/__nf_conntrack_hash_insert: netns: %u, status: %x\n", netns, status);
+
+    conn_tuple_t orig_conn = {};
+    if (!conntrack_tuple_to_conn_tuple(&orig_conn, &orig)) {
+        return 0;
+    }
+    orig_conn.netns = netns;
+
+    log_debug("orig\n");
+    print_translation(&orig_conn);
```

Comment:
I think this should be wrapped in a `#if DEBUG == 1`, or we could have a macro for it similarly to `log_debug`

Reply:
It ends up calling `log_debug`, so it should all get removed on non-debug builds. I will verify this.

Reply:
Confirmed. All traces to debug printing are gone.